src/components/Header.ts

```diff
@@ -1,24 +1,40 @@
 import ImagePath from '../../templates/add-button.png';
-import { $ } from '../util/dom';
+import { Component } from '../type/Component';
+
+class Header implements Component {
+  $target: Element;
+
+  constructor(parent: Element) {
+    parent.insertAdjacentHTML('afterbegin', this.template());
+    this.$target = parent.firstElementChild!;
+  }
 
-class Header {
   template = () => `
-    <header class="gnb">
-      <h1 class="gnb__title text-title">점심 뭐 먹지</h1>
-      <button type="button" class="gnb__button" aria-label="음식점 추가">
-        <img src= ${ImagePath} alt="음식점 추가">
-      </button>
+    <header>
+      <div class="gnb">
+        <h1 class="gnb__title text-title">점심 뭐 먹지</h1>
+        <button type="button" class="gnb__button" aria-label="음식점 추가">
+          <img src= ${ImagePath} alt="음식점 추가">
+        </button>
+      </div>
+      <div class="gnb__tab">
+          <input type="radio" name="tabs" id="all-restaurant" checked>
+          <label for="all-restaurant">모든 음식점</label>
+          <input type="radio" name="tabs" id="favorite-restaurant">
+          <label for="favorite-restaurant">자주 가는 음식점</label>
+      </div>
     </header>`;
 
-  render = (target: HTMLElement) => {
-    target.insertAdjacentHTML('afterbegin', this.template());
-  };
-
   setButtonHandler = (handler: () => void) => {
-    $('.gnb__button')?.addEventListener('click', (event) => {
+    this.$target.querySelector('.gnb__button')?.addEventListener('click', (event) => {
       handler();
     });
```

Comment:
```suggestion
    this.$target.querySelector('.gnb__button')?.addEventListener('click',  handler);
```

바로 함수레퍼런스를 전달해주어도 좋을 것 같아유